packages/opentelemetry-instrumentation-openai-agents/opentelemetry/instrumentation/openai_agents/_hooks.py

```diff
@@ -236,11 +236,17 @@ def on_span_end(self, span):
                     for i, message in enumerate(input_data):
                         if hasattr(message, 'role') and hasattr(message, 'content'):
                             otel_span.set_attribute(f"{SpanAttributes.LLM_PROMPTS}.{i}.role", message.role)
-                            otel_span.set_attribute(f"{SpanAttributes.LLM_PROMPTS}.{i}.content", message.content)
+                            content = message.content
+                            if not isinstance(content, str):
+                                content = json.dumps(content)
+                            otel_span.set_attribute(f"{SpanAttributes.LLM_PROMPTS}.{i}.content", content)
                         elif isinstance(message, dict):
                             if 'role' in message and 'content' in message:
                                 otel_span.set_attribute(f"{SpanAttributes.LLM_PROMPTS}.{i}.role", message['role'])
-                                otel_span.set_attribute(f"{SpanAttributes.LLM_PROMPTS}.{i}.content", message['content'])
+                                content = message['content']
+                                if isinstance(content, dict):
+                                    content = json.dumps(content)
+                                otel_span.set_attribute(f"{SpanAttributes.LLM_PROMPTS}.{i}.content", content)
```

Comment:
don't we need this JsonEncoder stuff we have in other instrumentations?

Reply:
no it's not needed for OAI